src/DataTypes/DataTypeEnum.h

```diff
@@ -45,13 +45,16 @@ class DataTypeEnum final : public IDataTypeEnum, public EnumValues<Type>
 
 private:
     std::string type_name;
+    std::string my_sql_type_name;
     static std::string generateName(const Values & values);
+    static std::string generateMySQLName(const Values & values);
 
 public:
     explicit DataTypeEnum(const Values & values_);
 
     std::string doGetName() const override { return type_name; }
     const char * getFamilyName() const override;
+    const char * getSQLCompatibleName() const override { return my_sql_type_name.c_str(); }
```

Comment:
Now returning const char * starts to be dangerous, let's change the interface of this function to return String.

Also, let's generate it on the fly - do not store it in the class's member.